apps/analysis/views.py

```diff
@@ -113,6 +115,30 @@ def get_queryset(self):
         )
 
 
+class DiscardedEntriesViewSet(viewsets.ModelViewSet):
+    serializer_class = DiscardedEntriesSerializer
+    permissions_classes = [permissions.IsAuthenticated]
+    filterset_class = DisCardedEntriesFilterSet
+
+    def get_queryset(self):
+        return DiscardedEntries.objects.filter(analysis_pillar=self.kwargs['analysis_pillar_id'])
+
+    def get_serializer_context(self):
+        return {
+            **super().get_serializer_context(),
+            'analysis_pillar_id': self.kwargs.get('analysis_pillar_id'),
+        }
+
+
+class PillarEntriesViewSet(EntryViewSet):
+
+    def get_queryset(self):
+        discarded_entries = DiscardedEntries.objects.filter(
+            analysis_pillar=self.kwargs['analysis_pillar_id']
+        ).values('entry')
+        return Entry.objects.exclude(id__in=discarded_entries)
```

Comment:
Make sure permissions are handled here. This might take care of the permission. But let's add a test to validate that too.
```suggestion
        return super().get_queryset().exclude(id__in=discarded_entries)
```